src/schemata/string/Base64.ts

```diff
@@ -52,5 +52,9 @@ export const base64: PB.Pattern = pipe(
  * @category Schema
  */
 export const Base64: Base64S = make(S =>
-  pipe(S.pattern(base64, 'Base64'), S.padRight(2, '='), S.brand<Base64Brand>())
+  pipe(
+    S.pattern(base64, 'Base64'),
+    S.padRight({ by: 'ExactLength', exactLength: s => s.length + (s.length % 2) }, '='),
```

Comment:
`% 4`

also above replace line 42 (`PB.anyNumber(),`) with `PB.between(2, 4),`

in base64, each character encodes 6 bits of data. but a byte is 8 bits, so
1 byte -> 8 bits -> 6bits + 2bits, 2 base64 characters + 2 padding
2 bytes -> 16 bits -> 6 + 6 + 4 bits, 3 base64 characters + 1 padding
3 bytes -> 24 bits = 6 bits x 4; 4 base64 characters with no padding

so the base64 encoding length is always divisible by 4 but you'll never have more than 2 padding characters

Reply:
I don't think that's right, becase base64 strings can have more than four non-equals characters

Reply:
yes, 3 bytes of data fits perfectly into 4 encoded characters

```
btoa('a')
'YQ=='
btoa('aa')
'YWE='
btoa('aaa')
'YWFh'
btoa('aaaa')
'YWFhYQ=='
```

Reply:
I don't think between 2, 4 would match that last case

Reply:
re: mod 4, I tried that originally, but it doesn't pad it properly.  The tests fail

Reply:
ah, sorry. I meant

```
PB.between(2, 4),
PB.subgroup,
PB.anyNumber(),
```

in regex terms: `([A-Za-z0-9+/]{2,4})*`

Reply:
> re: mod 4, I tried that originally, but it doesn't pad it properly. The tests fail

you need both things, otherwise you could generate a number of base64 characters = `1 mod 4`, and then 2 characters of padding doesn't make it evenly divisible by 4

Reply:
With mod 4 it fails at the arbitrary with a counterexample: `A=`.  And this is because length `A` = 1, and 1 + 1 mod 2 = 2.  And you can't have more than 2 equals as padding

Reply:
you'd never have a single base64 character. that's only 6 bits and we're encoding from data that is always a multiple of 8 bits

Reply:
Oh wait, I think I see it now.  If instead of replacing the `PB.anyNumber()` I add a subgroup and `between` right before it, it'll guarantee at least two characters... Let me try that

Reply:
Hmmm... nope that just hangs the test

Reply:
Okay, what I actually need is `atLeast` while also supporting empty strings

Reply:
I saw that happen before... I wonder if we need to put _some_ limit on the `star` and `plus` quantifiers?

https://github.com/jacob-alford/schemata-ts/blob/1.0.0-schema-first/src/PatternBuilder.ts#L357-L359

like if they're generating an array of 2 million arbitraries then I'd expect the tests to probably hang :sweat:

Reply:
Okay, summary:
- Need to find a way for `at least 2 or 0`
- modulus 4 is correct, but I also need `4 - length mod 4` too